### PR TITLE
Adds a guide to API docs.

### LINKED ISSUE #3481: Add guide to API docs

### PR DESCRIPTION
**Pull request by rhyolight**
fixes #3481 

Preview at http://rhyolight.github.io/nupic/ in the **Guide** section.

### CODE REVIEW

File: docs/source/guide-algorithms.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+There are several components to an HTM system:
+
+1. encoding data into SDR using [Encoders](encoders.html)
+1. passing encoded data through the [Spatial Pooler](spatial-pooler.html)
+1. running [Temporal Memory](temporal-memory.html) over the Spatial Pooler's active columns
+1. extracting predictions using a [Classifier](classifiers.html)
+1. extracting anomalies using
+
+Each of these components can be run independently of each other. The only communication between the are binary arrays (SDRs) representing cell activations. Spatial representations are maintained within the _proximal_ synapse permanences between the Spatial Pooler's columns and the input space. Temporal representations are maintained within the _distal_ synapse permanences between the cells in the Temporal Memory representation.
```
Comment (oxtopus) [inline docs/source/guide-algorithms.md]: Consider rewriting `binary arrays (SDRs)` as `sparse binary arrays (SDRs)`.

File: docs/source/guide-swarming.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+## Swarming
+
+> Swarming is a process that automatically determines the best model for a given dataset. By "best", we mean the model that most accurately produces the desired output. Swarming figures out which optional components should go into a model (encoders, spatial pooler, temporal pooler, classifier, etc.), as well as the best parameter values to use for each component.
+
+We have plans to replace the current swarming library with a more universal parameter search library. This codebase is currently unmaintained. We currently have no API documentation for it.
```
Comment (oxtopus) [inline docs/source/guide-swarming.md]: re:

> We have plans to replace the current swarming library...

Do we?  If so, it'd be great to include some links to issues or other documentation of those plans.

**Comment by rhyolight** [comment]
cc @JonnoFTW

File: docs/source/guide-swarming.md (hunk 2 above)
Comment (rhyolight) [inline docs/source/guide-swarming.md]: Do we have any public documents about it? I don't think we do. Do you think we should make the plan public?

**Pull request merged**